File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/CS/CsDeleteReaddTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

#if !SILVERLIGHT
using Db4oUnit;
using Db4oUnit.Extensions;
using Db4objects.Db4o.Config;
using Db4objects.Db4o.Ext;
using Db4objects.Db4o.Tests.Common.CS;

namespace Db4objects.Db4o.Tests.Common.CS
{
	public class CsDeleteReaddTestCase : Db4oClientServerTestCase
	{
		public class ItemParent
		{
		}

		public class Item : CsDeleteReaddTestCase.ItemParent
		{
			public string name;

			public Item(string name_)
			{
				name = name_;
			}
		}

		/// <exception cref="System.Exception"></exception>
		protected override void Configure(IConfiguration config)
		{
			config.GenerateUUIDs(ConfigScope.Globally);
			config.GenerateCommitTimestamps(true);
			config.ObjectClass(typeof(CsDeleteReaddTestCase.Item)).ObjectField("name").Indexed
				(true);
		}

		/// <exception cref="System.Exception"></exception>
		protected override void Store()
		{
			Store(new CsDeleteReaddTestCase.Item("one"));
		}

		public virtual void TestDeleteReadd()
		{
			IExtObjectContainer client1 = Db();
			IExtObjectContainer client2 = OpenNewSession();
			CsDeleteReaddTestCase.Item item1 = (CsDeleteReaddTestCase.Item)((CsDeleteReaddTestCase.Item
				)RetrieveOnlyInstance(client1, typeof(CsDeleteReaddTestCase.Item)));
			CsDeleteReaddTestCase.Item item2 = (CsDeleteReaddTestCase.Item)((CsDeleteReaddTestCase.Item
				)RetrieveOnlyInstance(client2, typeof(CsDeleteReaddTestCase.Item)));
			long idBeforeDelete = client1.GetID(item1);
			client1.Delete(item1);
			client1.Commit();
			client2.Store(item2);
			client2.Commit();
			client2.Close();
			CsDeleteReaddTestCase.Item item3 = ((CsDeleteReaddTestCase.Item)RetrieveOnlyInstance
				(client1, typeof(CsDeleteReaddTestCase.Item)));
			long idAfterUpdate = client1.GetID(item3);
			Assert.AreEqual(idBeforeDelete, idAfterUpdate);
			new FieldIndexAssert(typeof(CsDeleteReaddTestCase.Item), "name").AssertSingleEntry
				(FileSession(), idAfterUpdate);
		}

		public static void Main(string[] arguments)
		{
			new CsDeleteReaddTestCase().RunAll();
		}
	}
}
#endif // !SILVERLIGHT

File: db4oj.tests/src/com/db4o/db4ounit/common/cs/CsDeleteReaddTestCase.java
/* Copyright (C) 2008  Versant Inc.  http://www.db4o.com */

package com.db4o.db4ounit.common.cs;

import com.db4o.config.*;
import com.db4o.ext.*;

import db4ounit.*;
import db4ounit.extensions.*;


public class CsDeleteReaddTestCase extends Db4oClientServerTestCase {
    
    public static class ItemParent {
        
    }
    
    public static class Item extends ItemParent{
        
        public String name;
        
        public Item(String name_){
            name = name_;
        }
    }
    
    
    @Override
    protected void configure(Configuration config) throws Exception {
    	config.generateUUIDs(ConfigScope.GLOBALLY);
    	config.generateCommitTimestamps(true);
    	config.objectClass(Item.class).objectField("name").indexed(true);
    }
    
    protected void store() throws Exception {
        store(new Item("one"));
    }
    
    public void testDeleteReadd(){
        ExtObjectContainer client1 = db();
        ExtObjectContainer client2 = openNewSession();
        
        Item item1 = (Item) retrieveOnlyInstance(client1, Item.class);
        Item item2 = (Item) retrieveOnlyInstance(client2, Item.class);
        
        long idBeforeDelete = client1.getID(item1);
        
        client1.delete(item1);
        
        client1.commit();
        
        client2.store(item2);
        client2.commit();
        client2.close();
        
        Item item3 = retrieveOnlyInstance(client1, Item.class);
        final long idAfterUpdate = client1.getID(item3);
        
        Assert.areEqual(idBeforeDelete, idAfterUpdate);
        
        new FieldIndexAssert(Item.class, "name").assertSingleEntry(fileSession(), idAfterUpdate);
    }
    
    public static void main(String[] arguments) {
        new CsDeleteReaddTestCase().runAll();
    }

}
